### PR TITLE
ref(features): make flag entity strategy an enum instead of boolean

### PR DESCRIPTION
**Pull request by JoshFerge**
there have been several cases of folks not knowing why their feature flag isn't working, and it's because it's hard to know they need to pass in `True`. this PR changes the parameter to `default_manager.add` to be an enum with documentation, so hopefully going forward developers who need to add a flag can easily figure out which type they want.

Follow ups:
- modify our devdocs describing this change https://develop.sentry.dev/feature-flags/
- change the getsentry feature flags to use this enum, remove the backwards compatibility logic
- make it so there is no default parameter, and the type of flag handler must be passed in.

### CODE REVIEW

**Review by evanpurkhiser — APPROVED** [review]

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/features/base.py [hunk 1]
```diff
@@ -62,3 +63,14 @@ def __init__(self, name: str, user: User) -> None:
...
+
+
+class FeatureHandlerStrategy(Enum):
+    """
+    This controls whether the feature flag is evaulated statically,
+    or if it's managed by a feature flag service.
+    See https://develop.sentry.dev/feature-flags/
+    """
+
+    STATIC = 1
```
Comment (wedamija) [inline src/sentry/features/base.py:75]: Not all other feature flags are managed statically. We have lots that are managed via options that are dynamic but not flagr

Comment (JoshFerge) [reply src/sentry/features/base.py:75]: @wedamija good point. maybe `External` v.s. `Internal`? open to ideas

Comment (markstory) [reply src/sentry/features/base.py:75]: `Internal` and `Remote` 🤷 `External` works too.

File: src/sentry/features/base.py [hunk 2]
```diff
@@ -62,3 +63,14 @@ def __init__(self, name: str, user: User) -> None:
 
     def get_subject(self) -> User:
         return self.user
+
+
+class FeatureHandlerStrategy(Enum):
+    """
+    This controls whether the feature flag is evaulated statically,
```
Comment (dashed) [inline src/sentry/features/base.py:70]:
```suggestion
    This controls whether the feature flag is evaluated statically,
```

**Review by dashed — APPROVED** [review]

**Review by evanpurkhiser — APPROVED** [review]
I assume you'll cut these over?

**Comment by JoshFerge** [comment]
> I assume you'll cut these over?

yes, going to make a getsentry PR following then docs

**Review by dashed — APPROVED** [review]

File: src/sentry/features/base.py [hunk 3]
```diff
@@ -62,3 +63,14 @@ def __init__(self, name: str, user: User) -> None:
...
+
+class FeatureHandlerStrategy(Enum):
+    """
+    This controls whether the feature flag is evaluated statically,
+    or if it's managed by a remote feature flag service.
+    See https://develop.sentry.dev/feature-flags/
+    """
+
+    INTERNAL = 1
+    REMOTE = 2
```
Comment (evanpurkhiser) [inline src/sentry/features/base.py:76]: Would be nice if we had comments on these?

Comment (JoshFerge) [reply src/sentry/features/base.py:76]: will f/u

**Comment by JoshFerge** [comment]
last minute revert -- i'm sketched out by this. going to do in smaller increments.

**Pull request merged**